### PR TITLE
fix what project is used when building builtins

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/170

### CODE REVIEW

File: src/JuliaInterpreter.jl [hunk 1]
```diff
@@ -25,7 +25,7 @@ const BUILTIN_FILE = joinpath(@__DIR__, "builtins-julia$(Int(VERSION.major)).$(I
 @info "Generating builtins for this julia version..."
 gen_builtins_file = joinpath(@__DIR__, "generate_builtins.jl")
 # Run as separate command to prevent including the generate_builtins into the precompile cache
-run(`$(Base.julia_cmd()) --startup-file=no $gen_builtins_file`)
+run(`$(Base.julia_cmd()) --project=@. --startup-file=no $gen_builtins_file`)
```
Comment (fredrikekre) [inline src/JuliaInterpreter.jl]:
```suggestion
run(`$(Base.julia_cmd()) --project=$(joinpath(@__DIR__, "..") --startup-file=no $gen_builtins_file`)
```

Comment (KristofferC) [reply src/JuliaInterpreter.jl]: Thanks. I wonder if all this is worth it just to save caching the builtin generator... Oh well.

**Comment by fredrikekre** [comment]
You need to step up one level from the `src` directory.

**Comment by KristofferC** [comment]
Don't think so, `.@` will look upwards, no?

**Comment by fredrikekre** [comment]
Actually, I ran into this issue because I had a custom `LOAD_PATH` where `@` was not included, so the fix should be 
```diff
diff --git a/src/JuliaInterpreter.jl b/src/JuliaInterpreter.jl
index b2c6d43..2625fa1 100644
--- a/src/JuliaInterpreter.jl
+++ b/src/JuliaInterpreter.jl
@@ -25,7 +25,9 @@ const BUILTIN_FILE = joinpath(@__DIR__, "builtins-julia$(Int(VERSION.major)).$(I
 @info "Generating builtins for this julia version..."
 gen_builtins_file = joinpath(@__DIR__, "generate_builtins.jl")
 # Run as separate command to prevent including the generate_builtins into the precompile cache
-run(`$(Base.julia_cmd()) --startup-file=no $gen_builtins_file`)
+withenv("JULIA_LOAD_PATH"=joinpath(@__DIR__, "..")) do
+    run(`$(Base.julia_cmd()) --startup-file=no $gen_builtins_file`)
+end
 include_dependency(gen_builtins_file)
 
 include("types.jl")
```

File: src/JuliaInterpreter.jl [hunk 2]
```diff
@@ -25,7 +25,9 @@ const BUILTIN_FILE = joinpath(@__DIR__, "builtins-julia$(Int(VERSION.major)).$(I
 @info "Generating builtins for this julia version..."
 gen_builtins_file = joinpath(@__DIR__, "generate_builtins.jl")
 # Run as separate command to prevent including the generate_builtins into the precompile cache
-run(`$(Base.julia_cmd()) --startup-file=no $gen_builtins_file`)
+withenv("JULIA_LOAD_PATH"=joinpath(@__DIR__, "..")) do
```
Comment (fredrikekre) [inline src/JuliaInterpreter.jl]:
```suggestion
withenv("JULIA_LOAD_PATH" => joinpath(@__DIR__, "..")) do
```

**Comment by timholy** [comment]
Looks fine to me, but y'all know more about this part than I do.

**Pull request merged**